BaseLib/FileTools.cpp

```diff
@@ -39,6 +39,107 @@ bool IsFileExisting(const std::string &strFilename)
     return (stat (strFilename.c_str(), &buffer) == 0);
 }
 
+std::tuple<std::string, std::string::size_type, std::string::size_type>
+getParenthesizedString(std::string const& in,
+                       char const open_char,
+                       char const close_char,
+                       std::string::size_type pos)
+{
+    auto const pos_curly_brace_open = in.find_first_of(open_char, pos);
+    if (pos_curly_brace_open != std::string::npos)
+    {
+        auto const pos_curly_brace_close =
+            in.find_first_of(close_char, pos_curly_brace_open);
+        if (pos_curly_brace_close != std::string::npos)
+        {
+            return std::make_tuple(
+                in.substr(pos_curly_brace_open + 1,
+                          pos_curly_brace_close - (pos_curly_brace_open + 1)),
```

Comment:
Looks like a distance. Should be checked for being positive. *e.g.* "abc } def {"

Reply:
Since the search for the closing brace starts from `pos_curly_brace_open` the distance must be positive.